.github/workflows/build.yml

```diff
@@ -71,6 +71,9 @@ jobs:
         run: |
           set -eo pipefail
 
+          # disable cache mounts as github cache is slow
+          find -name Dockerfile -print0 | xargs -0 sed -Ei 's/--mount=type=cache,target=[^[:blank:]]+//g'
+
```

Comment:
I think we did that before, and removed it as it causes the pushed images to be from a different source than what's in the tree: it messes up the build cache if you just pull

Reply:
This is just a cherry-pick for convenience, the right PR is here https://github.com/OpenRailAssociation/osrd/pull/7991 ;)

You mean, people pulling CI-built images will not reuse the cached layer if they push? If so, I think that's an okay tradeoff.